### PR TITLE
Merge LCAO-refactor into develop

### PR DESCRIPTION
**Pull request by maki49**
(See #891 for detailed modification. )

### CODE REVIEW

File: source/module_esolver/esolver.h [hunk 1]
```diff
@@ -34,19 +34,17 @@ class ESolver
...
-    virtual void cal_Energy(energy &en) = 0; 
+    //Deal with exx and other calculation than scf/md/relax: 
+    // such as nscf, istate-charge or envelope
+    virtual void othercalculation(const int istep) {};
+
+    virtual void cal_Energy(energy& en) = 0;
     virtual void cal_Force(ModuleBase::matrix &force) = 0;
-    virtual void cal_Stress(ModuleBase::matrix &stress) = 0;
-    
+    virtual void cal_Stress(ModuleBase::matrix& stress) = 0;
```
Comment (dyzheng) [inline source/module_esolver/esolver.h:45]: Please format your code, using "shift"+ "alt" + "F" in vs-code for each changed file.

File: source/module_orbital/ORB_control.h [hunk 2]
```diff
@@ -79,18 +79,18 @@ class ORB_control
...
-    const int nbands = 0;
-    const int nspin = 1;
-    const int dsize = 0;
-    const int nb2d = 0;
-    const int dcolor = 0;
-    const int drank = 0;
-    const int myrank = 0;
-    const std::string calculation;
-    const std::string ks_solver;
+//private:
```
Comment (dyzheng) [inline source/module_orbital/ORB_control.h:82]: Are these variables Inevitably visited?

File: source/src_lcao/ELEC_nscf.h [hunk 3]
```diff
@@ -23,7 +23,7 @@ class ELEC_nscf
 	ELEC_nscf();
 	~ELEC_nscf();
 
-	private:
+	//private:
```
Comment (dyzheng) [inline source/src_lcao/ELEC_nscf.h:26]: Please move the target function or variable to public, rather than delete private directly.

Comment (dyzheng) [reply source/src_lcao/ELEC_nscf.h:26]: If some code outside module_esolver should be changed, you should add annotation to describe why these code should be changed.

**Review by dyzheng — CHANGES_REQUESTED** [review]

File: source/module_orbital/ORB_control.h (hunk 2 above)
Comment (maki49) [reply source/module_orbital/ORB_control.h:82]: They will only be visited in ESolver_KS_LCAO. I will add some comments in a new PR.

**Pull request closed without merging**